jaxrs/src/main/java/se/fortnox/reactivewizard/jaxrs/JaxRsRequestHandler.java

```diff
@@ -87,11 +92,16 @@ public JaxRsRequestHandler(Object[] services,
 
         long requestStartTime = System.currentTimeMillis();
 
-        return resource.call(jaxRsRequest)
-            .singleOrDefault(null)
-            .flatMap(result -> writeResult(response, result))
-            .onErrorResumeNext(e -> exceptionHandler.handleException(request, response, e))
-            .doAfterTerminate(() -> resource.log(request, response, requestStartTime));
+        try {
+            JaxRsRequestContext.open();
+            return resource.call(jaxRsRequest)
+                .singleOrDefault(null)
+                .flatMap(result -> writeResult(response, result))
+                .onErrorResumeNext(e -> exceptionHandler.handleException(request, response, e))
+                .doAfterTerminate(() -> resource.log(request, response, requestStartTime));
+        } finally {
+            JaxRsRequestContext.close();
```

Comment:
Move to doOnTerminate or it will be closed right after the observable is returned.
Some test for this would be nice.

Reply:
That is by design. I will add a test to show how it works.